docs/contributors/release.md

```diff
@@ -200,6 +180,28 @@ You should check that folks are able to install the new version from their Dashb
 
 If you don't have access to [make.wordpress.org/core](https://make.wordpress.org/core/), ping [someone on the Gutenberg Core team](https://github.com/orgs/WordPress/teams/gutenberg-core) in the [WordPress #core-editor Slack channel](https://wordpress.slack.com/messages/C02QB2JS7) to publish the post.
 
+## Packages Releases and WordPress Core Updates
+
+WordPress Core Updates are based on the `g-minor` branch. Releasing packages in order to update WordPress Core involves updating the `g-minor` branch (the workflow depends on whether it's a minor or major WordPress release) and run the package release process.
+
+### Major WordPress Releases
```

Comment:
I think the terminology is technically correct, but it might be confusing to someone who's unfamiliar that in the context of WordPress releases, a "major" release corresponds to what would be a minor version in SemVer, and a "minor" release corresponds to a patch version.